include/odp/api/spec/packet_io.h

```diff
@@ -1255,6 +1256,77 @@ typedef struct odp_pktio_info_t {
  */
 int odp_pktio_info(odp_pktio_t pktio, odp_pktio_info_t *info);
 
+/**
+ * Packet IO link speed in Mbps
+ */
+typedef enum odp_pktio_link_speed_t {
+	ODP_PKTIO_LINK_SPEED_UNKNOWN = 0,
+	ODP_PKTIO_LINK_SPEED_10M    = 10,
+	ODP_PKTIO_LINK_SPEED_100M   = 100,
+	ODP_PKTIO_LINK_SPEED_1G     = 1000,
+	ODP_PKTIO_LINK_SPEED_2_5G   = 2500,
+	ODP_PKTIO_LINK_SPEED_5G     = 5000,
+	ODP_PKTIO_LINK_SPEED_10G    = 10000,
+	ODP_PKTIO_LINK_SPEED_20G    = 20000,
+	ODP_PKTIO_LINK_SPEED_25G    = 25000,
+	ODP_PKTIO_LINK_SPEED_40G    = 40000,
+	ODP_PKTIO_LINK_SPEED_50G    = 50000,
+	ODP_PKTIO_LINK_SPEED_56G    = 56000,
+	ODP_PKTIO_LINK_SPEED_100G   = 100000,
+	ODP_PKTIO_LINK_SPEED_200G   = 200000,
+	ODP_PKTIO_LINK_SPEED_400G   = 400000
+} odp_pktio_link_speed_t;
+
+/** Autonegotiation mode */
+typedef enum odp_pktio_link_autoneg_t {
+	/** Autonegotiation disabled */
+	ODP_PKTIO_LINK_FIXED = 0,
+	/** Autonegotiation enabled */
+	ODP_PKTIO_LINK_AUTONEG  = 1
+} odp_pktio_link_autoneg_t;
```

Comment:
Instead of "fixed", these could be named "autoneg_off, autoneg_on", or something similar